### PR TITLE
[exporter/datasetexporter] Upgrade to dataset-go v0.19.0

### LINKED ISSUE #32533: [exporter/dataset] Integration stress tests failing
### Component(s)

exporter/dataset

### Describe the issue you're reporting

[Failing CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8743341054/job/23993767562?pr=32529)

Note that this test hasn't been running for some time, as explained in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32207, so this failure may not be the result of a recent change.

Failure output:
```
Running target 'mod-integration-test' in module 'exporter/datasetexporter' as part of group 'exporter-0'
make --no-print-directory -C exporter/datasetexporter mod-integration-test
running go integration test ./... in /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 360s -parallel 4 -tags=integration,""
∅  internal/metadata
✖  . (4.03s)

DONE 83 tests, 1 failure in 5.781s

✖  . (2.017s)

=== Failed
=== FAIL: . TestConsumeLogsManyLogsShouldSucceed (2.00s)
    logs_exporter_stress_test.go:92: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:92
        	Error:      	Received unexpected error:
        	            	cannot get DataSetExporter: cannot create newDatasetExporter: buffer settings are invalid: MaxLifetime 6225920 has to be at least 3 times smaller than PurgeOlderThan 0
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
    logs_exporter_stress_test.go:137: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:137
        	Error:      	Should be true
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
    logs_exporter_stress_test.go:140: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:140
        	Error:      	Not equal: 
        	            	expected: 0x30d40
        	            	actual  : 0x0
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
        	Messages:   	processed items
    logs_exporter_stress_test.go:141: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:141
        	Error:      	Not equal: 
        	            	expected: 0x30d40
        	            	actual  : 0x0
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
        	Messages:   	unique items

=== FAIL: . TestConsumeLogsManyLogsShouldSucceed (re-run 1) (2.00s)
    logs_exporter_stress_test.go:92: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:92
        	Error:      	Received unexpected error:
        	            	cannot get DataSetExporter: cannot create newDatasetExporter: buffer settings are invalid: MaxLifetime 62[259](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8743341054/job/23993767562?pr=32529#step:5:260)20 has to be at least 3 times smaller than PurgeOlderThan 0
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
    logs_exporter_stress_test.go:137: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:137
        	Error:      	Should be true
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
    logs_exporter_stress_test.go:140: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:140
        	Error:      	Not equal: 
        	            	expected: 0x30d40
        	            	actual  : 0x0
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
        	Messages:   	processed items
    logs_exporter_stress_test.go:141: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datasetexporter/logs_exporter_stress_test.go:141
        	Error:      	Not equal: 
        	            	expected: 0x30d40
        	            	actual  : 0x0
        	Test:       	TestConsumeLogsManyLogsShouldSucceed
        	Messages:   	unique items

DONE 2 runs, 84 tests, 2 failures in 8.602s
make[2]: *** [../../Makefile.Common:142: mod-integration-test] Error 1
make[1]: *** [Makefile:165: exporter/datasetexporter] Error 2
make: *** [Makefile:122: gointegration-test] Error 2
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
```

### PR DESCRIPTION
**Pull request by martin-majlis-s1**
**Description:** This PR is upgrading dataset-go to version v0.19.0 - https://github.com/scalyr/dataset-go/releases/tag/v0.19.0

This PR is also fixing:
* #33498 - config validation is not called during conversion, therefore I had to remove tests from logs and trace exporter. These tests are now moved to the factory.
* #32533 - fixing failing integration tests, that were never updated and therefore the configuration contained invalid values

**Link to tracking Issue:** #33498, #32533, #33675

**Testing:** Unit tests and integration tests.

**Documentation:** <Describe the documentation added.>

fixes #32533

### CODE REVIEW

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.21.11
+toolchain go1.22.4
```
Comment (crobert-1) [inline cmd/otelcontribcol/go.mod:7]:
```suggestion
toolchain go1.21.11
```
This is causing the GitHub action `make genotelcontribcol` failure. Your local environment is running the command on `go1.22.4`, while GitHub is running against `go1.21.11`. When the GitHub action generates this file it generates it with the older toolchain version. When you run it locally, it generates it with your local golang version.

File: .chloggen/datasetexporter-update-to-0.19.0.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'bug_fix'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datasetexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Upgrade dataset-go to v0.19.0 and fix found issues
```
Comment (tomaz-s1) [inline .chloggen/datasetexporter-update-to-0.19.0.yaml:10]: May want to briefly clarity what those "found issues" were under `subtext`.

**Review by tomaz-s1 — APPROVED** [review]
Small comment on expanding the changelog entry, besides that, LGTM.

Thanks.

**Comment by martin-majlis-s1** [comment]
> Small comment on expanding the changelog entry, besides that, LGTM.
> 
> Thanks.

I have added those details. My expectation was, that changelog is meant for users - and they do not care whether integration tests are now running when in the past they have been skipped.

**Review by zdaratom-s1 — APPROVED** [review]
LGTM

**Review by atoulme — APPROVED** [review]

**Pull request merged**